src/clarity-angular/emphasis/alert/providers/multi-alert-service.spec.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) 2017 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import { ComponentFixture, TestBed, fakeAsync, tick, inject } from "@angular/core/testing";
+import { QueryList } from "@angular/core";
+import { Alert } from "./../alert";
+import { ClrEmphasisModule } from "./../../emphasis.module";
+import { MultiAlertService } from "./multi-alert-service";
+
+describe("Multi Alert provider", function() {
+    let queryList: QueryList<Alert>;
+    let alert: ComponentFixture<Alert>;
+    let anotherAlert: ComponentFixture<Alert>;
+
+    beforeEach(function() {
+        TestBed.configureTestingModule({
+            imports: [ClrEmphasisModule],
+            providers: [MultiAlertService]
+        });
+
+        alert = TestBed.createComponent(Alert);
+        anotherAlert = TestBed.createComponent(Alert);
+
+        queryList = new QueryList<Alert>();
+        queryList.reset([
+            alert.componentInstance,
+            anotherAlert.componentInstance
+        ]);
+    });
+
+    it("next updates the current alert index and stays within bounds",
+        inject([MultiAlertService], function(multiAlertService: MultiAlertService) {
+            multiAlertService.manage(queryList);
+            expect(multiAlertService.current).toBe(0);
+            multiAlertService.next();
+            expect(multiAlertService.current).toBe(1);
+            multiAlertService.next();
+            expect(multiAlertService.current).toBe(1);
+    }));
+
+    it("previous method updates the current alert index and stays within bounds",
+        inject([MultiAlertService], function(multiAlertService: MultiAlertService) {
+            multiAlertService.manage(queryList);
+            multiAlertService.next();
+            expect(multiAlertService.current).toBe(1);
+            multiAlertService.previous();
+            expect(multiAlertService.current).toBe(0);
+            multiAlertService.previous();
+            expect(multiAlertService.current).toBe(0);
+    }));
+
+    it("close reduces the number of active alerts and updates the index",
+        fakeAsync(inject([MultiAlertService], function(multiAlertService: MultiAlertService) {
+            multiAlertService.manage(queryList);
+            multiAlertService.next();
+            expect(multiAlertService.count).toBe(2);
+            expect(multiAlertService.current).toBe(1);
+            anotherAlert.componentInstance.close();
+            tick();
```

Comment:
I might have missed something, but I don't remember any sources of asynchronicity in the code. Why do we need `fakeAsync()` and `tick()`?

Reply:
You didn't miss anything, it was remnant from something else I was trying. Removed.